src/Data/Nat/Show.agda

```diff
@@ -8,17 +8,78 @@
 
 module Data.Nat.Show where
 
-open import Data.Nat
+open import Data.Char as Char using (Char)
+open import Data.Digit using (showDigit; toDigits)
+open import Function   using (_∘_; _$_)
+open import Data.List  using (List; []; _∷_; map; reverse)
+open import Data.Nat   using (ℕ; _≟_; suc; pred; _+_; _*_; _<_; _>_; z≤n; s≤s;
+                                                                      _≤?_)
+open import Data.Nat.DivMod     using (_div_; _%_; a≡a%n+[a/n]*n; [a/n]*n≤a)
+open import Data.Product        using (proj₁)
+open import Data.Nat.Properties using
+                 (+-comm; *-identityʳ; <-transˡ; <-irrefl; <⇒≢; n≢0⇒n>0;
+                  m≤m+n; *-monoʳ-<; m<m*n; m≢0⇒suc[pred[m]]≡m
+                 )
+open import Data.String as String using (String)
+open import Induction.Nat         using (<-wellFounded)
+open import Induction.WellFounded using (Acc; acc)
+open import Relation.Binary.PropositionalEquality using (_≡_; _≢_; refl; sym;
+                                                         cong; ≢-sym)
+open import Relation.Nullary           using (yes; no)
 open import Relation.Nullary.Decidable using (True)
-open import Data.String.Base as String using (String)
-open import Data.Digit
-open import Data.Product using (proj₁)
-open import Function
-open import Data.List.Base
+open import Relation.Nullary.Negation  using (contradiction)
 
--- showInBase b n is a string containing the representation of n in
--- base b.
 
+------------------------------------------------------------------------------
+-- Conversion from unary representation to the representation by the given
+-- base.
+toDigitNats : (base : ℕ) → base > 1 → ℕ → List ℕ
+toDigitNats 0             ()
+toDigitNats 1             (s≤s ())
+toDigitNats (suc (suc b)) base>1 x =  aux x (<-wellFounded x) []
```

Comment:
The following names are not used anymore and can now be removed:
```
n'≡r+q*base
pq
pbase
suc-pq≡q
```
The name `base` can be better expressed using the `@` syntax. If in the line above you replace `(suc (suc b))` with `base@(suc (suc b))`